awx/ui_next/src/screens/Job/JobOutput/HostEventModal.jsx

```diff
@@ -0,0 +1,218 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Button,
+  Modal as PFModal,
+  Tab,
+  Tabs as PFTabs,
+} from '@patternfly/react-core';
+import CodeMirrorInput from '@components/CodeMirrorInput';
+import ContentEmpty from '@components/ContentEmpty';
+import { DetailList, Detail } from '@components/DetailList';
+import { HostStatusIcon } from '@components/Sparkline';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import styled from 'styled-components';
+import Entities from 'html-entities';
+
+const entities = new Entities.AllHtmlEntities();
+
+const Modal = styled(PFModal)`
+  --pf-c-modal-box__footer--MarginTop: 0;
+  .pf-c-modal-box__body {
+    overflow-y: hidden;
+  }
+  .pf-c-tab-content {
+    padding: 24px 0;
+  }
+`;
+
+const HostNameDetailValue = styled.div`
+  align-items: center;
+  display: inline-grid;
+  grid-gap: 10px;
+  grid-template-columns: min-content auto;
+`;
+
+const Tabs = styled(PFTabs)`
+  --pf-c-tabs__button--PaddingLeft: 20px;
+  --pf-c-tabs__button--PaddingRight: 20px;
+
+  .pf-c-tabs__list {
+    li:first-of-type .pf-c-tabs__button {
+      &::after {
+        margin-left: 0;
+      }
+    }
+  }
+
+  &:not(.pf-c-tabs__item)::before {
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    content: '';
+    border-bottom: solid var(--pf-c-tabs__item--BorderColor);
+    border-width: var(--pf-c-tabs__item--BorderWidth) 0
+      var(--pf-c-tabs__item--BorderWidth) 0;
+  }
+`;
+
+function HostEventModal({ handleClose, hostEvent, isOpen, i18n }) {
+  const [hostStatus, setHostStatus] = useState(null);
+  const [activeTabKey, setActiveTabKey] = useState(0);
+
+  useEffect(() => {
+    processEventStatus(hostEvent);
+  }, []);
+
+  const handleTabClick = (event, tabIndex) => {
+    setActiveTabKey(tabIndex);
+  };
+
+  function processEventStatus(event) {
+    let status = null;
+    if (event.event === 'runner_on_unreachable') {
+      status = 'unreachable';
+    }
+    // equiv to 'runner_on_error' && 'runner_on_failed'
+    if (event.failed) {
+      status = 'failed';
+    }
+    // catch the 'changed' case before 'ok', because both can be true
+    if (event.changed) {
```

Comment:
You have the right idea about it being possible for both to be true. However, I think we want to reverse the order of these checks.

If an event qualifies as changed, we _always_ want to represent it as such, even if it also qualifies as `ok`. This looks like an `ok` host status will always override `changed`.

Reply:
This logic was pulled from the host-event angular service - should I go ahead and update this in both locations? Or just within the scope of this PR?

Reply:
Yes - let's fix it there too. We can use a separate PR or include it in this one, doesn't matter to me.

Reply:
The change was added to this PR https://github.com/ansible/awx/pull/4677/commits/e53c9793441b4eb80525d9f1bf16e0ebe0d7aa74

Reply:
The change was added to this PR https://github.com/ansible/awx/pull/4677/commits/e53c9793441b4eb80525d9f1bf16e0ebe0d7aa74